### PR TITLE
Add gov support to bnscli

### LINKED ISSUE #772: Add governance to cmd/bnscli
As [discussed briefly](https://internetofvalues.slack.com/archives/GFASTCVPZ/p1560766927011000) as a governance member, I would like to use `cmd/bnscli` to access `x/gov` functionality so that I can participate in the governance process.

The current `bnscli` version contains support [for the proposal creation](https://github.com/iov-one/weave/blob/master/cmd/bnscli/cmd_gov.go#L20). Support for other operations must be added.

## Acceptance criteria
- [x] add support in `cmd/bnsd` for creating the [`DeleteProposalMsg`](https://github.com/iov-one/weave/blob/master/x/gov/codec.proto#L202),
- [x] add support in `cmd/bnsd` for creating the [`VoteMsg`](https://github.com/iov-one/weave/blob/master/x/gov/codec.proto#L217),
- [x] add support in `cmd/bnsd` for creating the [`TallyMsg`](https://github.com/iov-one/weave/blob/master/x/gov/codec.proto#L230),
- [x] add support in `cmd/bnsd` for creating the [`TextResolutionMsg`](https://github.com/iov-one/weave/blob/master/x/gov/codec.proto#L239),
- [x] add support in `cmd/bnsd` for creating the [`UpdateElectorateMsg`](https://github.com/iov-one/weave/blob/master/x/gov/codec.proto#L244),
- [x] add support in `cmd/bnsd` for creating the [`UpdateElectionRuleMsg`](https://github.com/iov-one/weave/blob/master/x/gov/codec.proto#L253),

### PR DESCRIPTION
**Pull request by alpe**
Adds gov functionality to bnscli to
- delete a proposal
- vote
- tally

See #772

### CODE REVIEW

File: cmd/bnscli/cmd_gov.go [hunk 1]
```diff
@@ -257,103 +257,31 @@ func inOneHour() time.Time {
...
+Delete an existing proposal before the voting period has started.
+		`)
+		fl.PrintDefaults()
+	}
+	var (
+		id = flSeq(fl, "id", "", "The ID of the proposal that is to be deleted.")
+	)
+	fl.Parse(args)
+	if len(*id) == 0 {
+		return errors.New("the id must not be empty")
```
Comment (husio) [inline cmd/bnscli/cmd_gov.go]: You can also `flagDie`. Should not make much difference.

**Comment by iov-bot-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry. <br />
You can find it at <a href="https://github.com/iov-one/weave/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: cmd/bnscli/cmd_gov.go [hunk 2]
```diff
@@ -257,103 +257,31 @@ func inOneHour() time.Time {
...
+	}
+	gov.NewProposalBucket()
```
Comment (husio) [inline cmd/bnscli/cmd_gov.go]: :thinking: Is there a side effect to bucket creation?

File: cmd/bnscli/cmd_gov.go [hunk 3]
```diff
@@ -257,103 +257,31 @@ func inOneHour() time.Time {
 	return time.Now().Add(time.Hour)
 }
 
-/*
-#!/bin/bash
```
Comment (husio) [inline cmd/bnscli/cmd_gov.go]: This comment is a great time saver when transaction declaration is changing. Did you intend to move it to a separate file and forgot to add the file?

Comment (alpe) [reply cmd/bnscli/cmd_gov.go]: no, this came without a comment so I assumed it is dead code. Can you elaborate what it does and why we should keep it (in a separate file probably?), please

File: cmd/bnscli/cmd_gov.go (hunk 2 above)
Comment (alpe) [reply cmd/bnscli/cmd_gov.go]: will remove. this: 🐍 ed in

File: cmd/bnscli/cmd_gov.go (hunk 3 above)
Comment (husio) [reply cmd/bnscli/cmd_gov.go]: Maybe it's time to make it a separate file.

This is a script for generating Go code https://github.com/iov-one/weave/blob/master/cmd/bnscli/cmd_gov.go#L51-L60
It saves a lot of time (and bugs) creating type casting and conversion for messages used by `bnsd` transactions.

Comment (alpe) [reply cmd/bnscli/cmd_gov.go]: extracted to a new file for doc not execution

**Pull request merged**